plugins/lua/ts_lua.c

```diff
@@ -68,12 +79,18 @@ TSRemapNewInstance(int argc, char *argv[], void **ih, char *errbuf, int errbuf_s
   char *inline_script                  = "";
   int fn                               = 0;
   int states                           = TS_LUA_MAX_STATE_COUNT;
+  TSMgmtInt mgmt_states                = 0;
   static const struct option longopt[] = {
     {"states", required_argument, 0, 's'},
     {"inline", required_argument, 0, 'i'},
     {0, 0, 0, 0},
   };
 
+  if (TS_SUCCESS == TSMgmtIntGet(ts_lua_mgmt_state_str, &mgmt_states) && 0 < mgmt_states) {
```

Comment:
one more condition should be that mgmt_states cannot be larger than TS_LUA_MAX_STATE_COUNT

Reply:
I think we have checks afterwards to make sure of that. So it should be fine